check_mariadb_slaves.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+"""MariaDB slave status checker"""
+import sys
+import argparse
+import MySQLdb
+
+
+class NagiosPlugin(object):
+    """Nagios Plugin base class"""
+
+    def __init__(self, warning, critical, *args, **kwargs):
+        self.warning = warning
+        self.critical = critical
+
+    def run_check(self):
+        raise NotImplementedError
+
+    def ok_state(self, msg):
+        print "OK - {}".format(msg)
+        sys.exit(0)
+
+    def warning_state(self, msg):
+        print "WARNING - {}".format(msg)
+        sys.exit(1)
+
+    def critical_state(self, msg):
+        print "CRITICAL - {}".format(msg)
+        sys.exit(2)
+
+    def unknown_state(self, msg):
+        print "UNNKNOWN - {}".format(msg)
+        sys.exit(3)
+
+
+class SlaveStatusCheck(NagiosPlugin):
+    """Class to help us run slave status queries against MariaDB"""
+    REPLICATION_LAG_MODE = 'replication_lag'
+    SLAVESQL_MODE = 'slave_sql'
+    SLAVEIO_MODE = 'slave_io'
+    MODES = (REPLICATION_LAG_MODE,
+             SLAVESQL_MODE,
+             SLAVEIO_MODE)
+
+    def __init__(self, hostname, username, password, connection_name,
+                 mode, verbose=False, warning=None, critical=None):
+        super(SlaveStatusCheck, self).__init__(warning, critical)
+
+        self.hostname = hostname
+        self.username = username
+        self.password = password
+        self.connection_name = connection_name
+        self.verbose = verbose
+        self.mode = mode
+        self._slave_status = {}
```

Comment:
Validate that the mode passed is in the set of MODES - otherwise, self.unknown_state("Invalid mode")